src/super_gradients/training/utils/media/image.py

```diff
@@ -50,17 +51,29 @@ def generate_image_loader(images: Union[List[ImageSource], ImageSource]) -> Iter
         images_paths = list_images_in_folder(images)
         for image_path in images_paths:
             yield load_image(image=image_path)
-    elif _is_batch_of_images(images=images):
+    elif _is_4d_array(images):
+        warnings.warn(
+            "It seems you are using predict() with 4D array as input. "
+            "Please note we cannot track whether the input was already normalized or not. "
+            "You will get incorrect results if you feeding batches from train/validation dataloader."
```

Comment:
typo: if you are*

Reply:
Fixed in 1337b83b832831704bdcdc23b651c39c1af81027